build.yaml

```diff
@@ -251,42 +251,15 @@ common_cli:
     # Set up the new venv.
     virtualenv -p python2.7 --clear ${venv}
     . ${venv}/bin/activate
-    # Report the version of Python we're using, to aid debugging.
+    # Report the version of Python and pip we're using, to aid debugging.
     ${venv}/bin/python --version
-
-  setup_pip_cache: &setup_pip_cache |
-    # exports the PIP_INDEX_URL, TRUSTED_HOST variables to the environment.
-    # The /tmp/pip.sh file is copied to the Jenkins Slave by the Jenkins Master
-    # during the bootstrapping of the slave.
-    # This file is located in /etc/jenkins_slave on the Master box.
-    # These variables contain the PIP URL and IP address of our caching server.
-    if [ -e /tmp/pip.sh ]; then
-      . /tmp/pip.sh
-    fi
+    ${venv}/bin/pip install --upgrade pip
+    ${venv}/bin/pip --version
 
   setup_flocker_modules: &setup_flocker_modules |
     # installs all the required python modules as well as the flocker code.
-    # But first, we need to upgrade pip to 7.1.
-    # We have a devpi cache in AWS which we will consume instead of going
-    # upstream to the PyPi servers.
-    # We specify that devpi caching server using -i $PIP_INDEX_URL
-    # which requires as to include --trusted-host as we are not (yet) using
-    # SSL on our caching box.
-    # The --trusted-host option is only available with pip 7.
-    #
-    if [ ${PIP_INDEX_URL} ]; then
-        PIP_ADDITIONAL_OPTIONS="${PIP_ADDITIONAL_OPTIONS} -i ${PIP_INDEX_URL} "
-    fi
-    if [ ${TRUSTED_HOST} ]; then
-        PIP_ADDITIONAL_OPTIONS="${PIP_ADDITIONAL_OPTIONS} \
-                                --trusted-host ${TRUSTED_HOST} "
-    fi
-
     # using the caching-layer, install all the dependencies
-    pip install --upgrade pip
-    # Use --process-dependency-links so we can use the pyrsistent fork.
-    pip install --process-dependency-links . ${PIP_ADDITIONAL_OPTIONS}
-    pip install --process-dependency-links ".[dev]" ${PIP_ADDITIONAL_OPTIONS}
+    pip install --requirement dev-requirements.txt ${PIP_ADDITIONAL_OPTIONS}
```

Comment:
I could probably ditch `PIP_ADDITIONAL_OPTIONS` here. I don't think anyone ever sets it.